packages/flutter_tools/templates/plugin/ios.tmpl/Classes/pluginClass.m.tmpl

```diff
@@ -1,25 +1,20 @@
 #import "{{pluginClass}}.h"
 
-@implementation {{pluginClass}} {
+@implementation {{pluginClass}}
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
+  FlutterMethodChannel* channel = [FlutterMethodChannel
+      methodChannelWithName:@"{{projectName}}"
+            binaryMessenger:[registrar messenger]];
+  {{pluginClass}}* instance = [{{pluginClass}} new];
```

Comment:
I have tried to avoid `new` as much as possible and write the more verbose `alloc` - `init` incantation to be more explicit. If you don't mind, can we stick to using it here and elsewhere? If you have in any way a strong opinion for `new` however, then go ahead and stick to it. At this point, it is more of a personal preference really.